alchemist-full/build.gradle.kts

```diff
@@ -33,23 +32,44 @@ plugins {
     alias(libs.plugins.shadowJar)
 }
 
-dependencies {
-    runtimeOnly(rootProject)
-    rootProject.subprojects.filterNot { it == project }.forEach {
-        if (it.isMultiplatform) {
-            runtimeOnly(project(path = ":${it.name}", configuration = "default"))
-        } else {
-            runtimeOnly(it)
+kotlin {
+    jvm {
+        withJava()
+    }
+    sourceSets {
+        val commonMain by getting {
+            dependencies {
+                runtimeOnly(rootProject)
+                rootProject.subprojects.filterNot { it == project }.forEach {
+                    runtimeOnly(it)
+                }
+            }
+        }
+        val commonTest by getting {
+            dependencies {
+                implementation(rootProject)
+                implementation(alchemist("physics"))
+            }
         }
     }
-    testImplementation(rootProject)
-    testImplementation(alchemist("physics"))
 }
 
 application {
     mainClass.set("it.unibo.alchemist.Alchemist")
 }
 
+/**
+ * Add the runtime classpath of the multiplatform JVM projects to the run task
+ */
+tasks.named("run", JavaExec::class) {
```

Comment:
Task references by name are very fragile, can't you use something like `tasks.run`?

Reply:
Solved, I also introduced a function to build the `FileCollection`and used it to both configure the `run` and `shadowJar` tasks. Take a look and notify me if something is out of place.